Added Framework Authentication (Issue #53).

DarinJ opened the pull request:
Modified `MesosScheduler.java` and `configuration.md`.  Now `mapred.mesos.framework.principal`, `mapred.mesos.framework.secretfile`, `mapred.mesos.framework.user`, and `mapred.mesos.framework.name` are configureable options.

tarnfeld commented:
Looks awesome, thanks for the contribution! I'll give it a whirl and merge if it looks good.

tarnfeld commented on src/main/java/org/apache/hadoop/mapred/MesosScheduler.java:
```diff
@@ -181,15 +183,36 @@ public synchronized void start() throws IOException {
+      Credential credential=null;
+
+      String frameworkPrincipal=conf.get("mapred.mesos.framework.principal");
+      if (frameworkPrincipal!=null){
```
Would you mind just tidying up this block of code you've added to be consistent with the rest of the file (basically just some additional spaces around your `if` statements and `=` signs.

Thanks!

tarnfeld commented:
Would you mind rebasing this against master? Sorry it's a pain I know!

DarinJ commented:
No issues rebasing, but there's another PR in the queue.  Which order do you want to merge them?  I can probably do the rebase in the next hour, but can wait if you'd rather merge the other in first.

tarnfeld commented:
The other PR is mine and isn't quite ready to go yet, so i'll sort that out. This one will definitely be merged first :smiley:

DarinJ commented:
Started rebase, more merge conflicts than expected.  Shouldn't be hard but don't want to rush it before a meeting expect done by EOD tomorrow.

DarinJ commented:
@tarnfeld rebased.  Tested locally to make sure the merge went fine.  Cheers.

PS: If you want a maven to enforce a checkstyle I could probably add it in the next week or so.

tarnfeld commented on src/main/java/org/apache/hadoop/mapred/MesosScheduler.java:
```diff
@@ -364,10 +366,35 @@ public synchronized void start() throws IOException {
+              .setName(conf.get("mapred.mesos.framework.name", "Hadoop: (RPC port: " + jobTracker.port + ","
+                      + " WebUI port: " + jobTracker.infoPort + ")"));
+
+      Credential credential=null;
```
Little bit of missing whitespace here

tarnfeld commented on src/main/java/org/apache/hadoop/mapred/MesosScheduler.java:
```diff
@@ -364,10 +366,35 @@ public synchronized void start() throws IOException {
+      } else {
+        LOG.info("Creatingg Schedule Driver, attempting to authenticate with Principal: " + credential.getPrincipal()
+                + ", secret:" + credential.getSecret());
+        driver = new MesosSchedulerDriver(this, frameworkInfoBuilder.build(), master,credential);
```
Missing a space after the comma

tarnfeld commented:
Just two teeny style comments, I think we should definitely add some tools for doing checkstyle or something like that. I'm just keen to keep the code consistent across the project at the moment.

If you fix those two things i'll merge the PR right away! Thanks again.

DarinJ commented:
@tarnfeld you caught me at a good time.  Updated and squashed commits.

I'll open an issue on checkstyles, I've got a few other things I'd like to contribute in the near future. Getting checkstyles will end up saving me time :).
